code/modules/projectiles/guns/smartgun.dm

```diff
@@ -551,37 +551,80 @@
 		if(linked_human.is_revivable() || linked_human.stat != DEAD)
-			to_chat(user, SPAN_WARNING("[icon2html(src)] Trigger locked by [src]. Unauthorized user."))
+			to_chat(user, SPAN_WARNING("[icon2html(src, usr)] Trigger locked by [src]. Unauthorized user."))
 			playsound(loc,'sound/weapons/gun_empty.ogg', 25, 1)
 			return FALSE
 
 		linked_human = null
 		is_locked = FALSE
 		UnregisterSignal(linked_human, COMSIG_PARENT_QDELETING)
 
-/obj/item/weapon/gun/smartgun/co/pickup(user)
-	if(!linked_human)
-		src.name_after_co(user, src)
-		to_chat(usr, SPAN_NOTICE("[icon2html(src)] You pick up \the [src], registering yourself as its owner."))
-	..()
+// ID lock action \\
+
+/datum/action/item_action/co_sg/action_activate()
+	var/obj/item/weapon/gun/smartgun/co/protag_gun = holder_item
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/protagonist = owner
+	if(protagonist.is_mob_incapacitated() || protag_gun.get_active_firearm(protagonist, FALSE) != holder_item)
+		return
+
+/datum/action/item_action/co_sg/update_button_icon()
+	return
 
-/obj/item/weapon/gun/smartgun/co/verb/toggle_lock()
-	set category = "Weapons"
-	set name = "Toggle Lock"
-	set src in usr
+/datum/action/item_action/co_sg/toggle_id_lock/New(Target, obj/item/holder)
+	. = ..()
+	name = "Toggle ID lock"
+	action_icon_state = "id_lock_locked"
+	button.name = name
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/hud/actions.dmi', button, action_icon_state)
 
-	if(usr != linked_human)
-		to_chat(usr, SPAN_WARNING("[icon2html(src)] Action denied by \the [src]. Unauthorized user."))
+/datum/action/item_action/co_sg/toggle_id_lock/action_activate()
+	. = ..()
+	var/obj/item/weapon/gun/smartgun/co/protag_gun = holder_item
+	protag_gun.toggle_lock()
+	if(protag_gun.is_locked)
+		action_icon_state = "id_lock_locked"
+	else
+		action_icon_state = "id_lock_unlocked"
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/hud/actions.dmi', button, action_icon_state)
+
+/obj/item/weapon/gun/smartgun/co/proc/toggle_lock(mob/user)
+	if(linked_human && usr != linked_human)
+		to_chat(usr, SPAN_WARNING("[icon2html(src, usr)] Action denied by \the [src]. Unauthorized user."))
 		return
+	else if(!linked_human)
+		name_after_co(usr)
 
 	is_locked = !is_locked
-	to_chat(usr, SPAN_NOTICE("[icon2html(src)] You [is_locked? "lock": "unlock"] \the [src]."))
+	to_chat(usr, SPAN_NOTICE("[icon2html(src, usr)] You [is_locked? "lock": "unlock"] \the [src]."))
 	playsound(loc,'sound/machines/click.ogg', 25, 1)
 
+// action end \\
+
+/obj/item/weapon/gun/smartgun/co/pickup(user)
+	if(!linked_human)
+		src.name_after_co(user, src)
+		to_chat(usr, SPAN_NOTICE("[icon2html(src, usr)] You pick up \the [src], registering yourself as its owner."))
+	..()
+
+
 /obj/item/weapon/gun/smartgun/co/proc/name_after_co(var/mob/living/carbon/human/H, var/obj/item/weapon/gun/smartgun/co/I)
 	linked_human = H
 	RegisterSignal(linked_human, COMSIG_PARENT_QDELETING, .proc/remove_idlock)
```

Comment:
Github won't let me go down any further but get_examine_text() (line 632) needs to be converted to the new system as it is only sending the to_chat() about registered user. See M46C for example.